datafusion/physical-plan/src/lib.rs

```diff
@@ -455,6 +443,22 @@ impl ExecutionPlanProperties for Arc<dyn ExecutionPlan> {
     ///
     /// See also [`ExecutionPlan::maintains_input_order`] and [`Self::output_ordering`]
     /// for related concepts.
+    fn equivalence_properties(&self) -> &EquivalenceProperties;
+}
+
+impl ExecutionPlanProperties for Arc<dyn ExecutionPlan> {
```

Comment:
```suggestion
impl ExecutionPlanProperties for dyn ExecutionPlan {
```
Maybe doing this will make the extension trait a bit more generally applicable?

Reply:
Added in 839d783c7

It turns out I had to also leave the 

```
impl ExecutionPlanProperties for Arc<dyn ExecutionPlan> {
```

Otherwise the compiler didn't seem to be able to find the relevant methods without code changes (e.g. `execution_plan.as_ref()....`)